File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/AddressForm.ts
import { StringEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { StateProvinceDialog } from "@/Company/StateProvince/StateProvinceDialog";
import { initFormType } from "@serenity-is/corelib/q";

export interface AddressForm {
    AddressLine1: StringEditor;
    AddressLine2: StringEditor;
    CountryCode: LookupEditor;
    StateProvinceId: LookupEditor;
    City: LookupEditor;
    PostalCode: StringEditor;
}

export class AddressForm extends PrefixedContext {
    static formKey = 'Company.Address';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AddressForm.init)  {
            AddressForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(AddressForm, [
                'AddressLine1', w0,
                'AddressLine2', w0,
                'CountryCode', w1,
                'StateProvinceId', w1,
                'City', w1,
                'PostalCode', w0
            ]);
        }
    }
}

[StateProvinceDialog]; // referenced types

File: Ser2023/Company/Company.Web/Modules/Company/Address/AddressForm.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.Address")]
[BasedOnRow(typeof(AddressRow), CheckNames = true)]
public class AddressForm
{
    [Category("Address")]
    [Width(100)]
    public string AddressLine1 { get; set; }
    [Width(100)]
    public string AddressLine2 { get; set; }
    [Category("Country, Province, City")]
    [TwoThirdWidth]
    public string CountryCode {  get; set; }
    [OneThirdWidth]
    public int StateProvinceId { get; set; }
    [TwoThirdWidth]
    public string City { get; set; }
    [OneThirdWidth]
    public string PostalCode { get; set; }
    //public Guid Rowguid { get; set; }
}